pkg/api/pod/util.go

```diff
@@ -326,3 +336,13 @@ func DropDisabledVolumeDevicesAlphaFields(podSpec *api.PodSpec) {
 		}
 	}
 }
+
+// DropDisabledCSIVolumeSourceAlphaFields removes disabled alpha fields from []CSIVolumeSource.
+// This should be called from PrepareForCreate/PrepareForUpdate for all pod specs resources containing a CSIVolumeSource
+func DropDisabledCSIVolumeSourceAlphaFields(podSpec *api.PodSpec) {
```

Comment:
@liggitt I have seen some PRs that drop alpha fields and others that fail validation if they are set.  Do we have a rule on this?

Reply:
>  I have seen some PRs that drop alpha fields and others that fail validation if they are set. Do we have a rule on this?

The guidelines in https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#adding-unstable-features-to-stable-versions call for both. The validation failure is redundant if the pre-create/update drops are done, but is a belt/suspenders check to ensure no alpha data is allowed.